### PR TITLE
Five Rerankers Added

### PR DESCRIPTION
**Pull request by ashwinzyx**
['cohere','jina','mixedbread-ai/mxbai-rerank-large-v1','flashrank','colbert']

### CODE REVIEW

File: src/ragbuilder/ragbuilder.py [hunk 1]
```diff
@@ -40,7 +40,7 @@
 LOG_DIRNAME = Path(LOG_FILENAME).parent
 BASE_DIR = Path(__file__).resolve().parent
 logger.info(f"LOG_FILENAME = {LOG_FILENAME}")
-
+os.environ['KMP_DUPLICATE_LIB_OK']='True'
```
Comment (aravind10x) [inline src/ragbuilder/ragbuilder.py:43]: What's the purpose of this?

**Review by aravind10x — CHANGES_REQUESTED** [review]

File: src/ragbuilder/langchain_module/retriever/retriever.py
Comment (aravind10x) [inline src/ragbuilder/langchain_module/retriever/retriever.py:1]: Need to update line 40 as well to retrieve 100 docs from retriever when reranking is involved.

File: src/ragbuilder/langchain_module/retriever/retriever.py [hunk 2]
```diff
@@ -139,7 +140,63 @@ def getCompressors(**kwargs):
         code_string= f"""arr_comp.append(LongContextReorder())"""
         import_string = f"""from langchain_community.document_transformers import LongContextReorder"""
         return {'code_string':code_string,'import_string':import_string}
-        
+    
+    if 'mixedbread-ai/mxbai-rerank-large-v1' in compressor_config:
+        code_string= f"""ranker = Reranker("mixedbread-ai/mxbai-rerank-base-v1", verbose=0)
```
[the remark this answers is not in the document]
Comment (ashwinzyx) [reply src/ragbuilder/langchain_module/retriever/retriever.py]: fixed

File: src/ragbuilder/ragbuilder.py (hunk 1 above)
Comment (ashwinzyx) [reply src/ragbuilder/ragbuilder.py]: removed. Not needed anymore

File: src/ragbuilder/langchain_module/retriever/retriever.py [hunk 3]
```diff
@@ -139,7 +140,63 @@ def getCompressors(**kwargs):
...
+    
+    if 'rankllm' in compressor_config:
+        code_string= f"""ranker = Reranker("rankllm",api_key = os.getenv('OPENAI_API_KEY'))
+compressor = ranker.as_langchain_compressor(k={search_kwargs})
+arr_comp.append(compressor)
+"""
+        import_string = f"""from rerankers import Reranker"""
+        return {'code_string':code_string,'import_string':import_string}
+
     if 'CrossEncoderReranker' in compressor_config:
```
[the remark this answers is not in the document]
Comment (ashwinzyx) [reply src/ragbuilder/langchain_module/retriever/retriever.py]: modified to BAAI/bge-reranker-base

File: src/ragbuilder/langchain_module/retriever/retriever.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (ashwinzyx) [reply src/ragbuilder/langchain_module/retriever/retriever.py]: fixed typo

File: src/ragbuilder/langchain_module/retriever/retriever.py
Comment (ashwinzyx) [reply src/ragbuilder/langchain_module/retriever/retriever.py:1]: Added to list

**Review by aravind10x — APPROVED** [review]

**Pull request merged**